remote/remote_test.go

```diff
@@ -1500,4 +1500,82 @@ func testImage(t *testing.T, when spec.G, it spec.S) {
 			})
 		})
 	})
+
+	when("#CheckReadAccess", func() {
+		when("image exists in the registry and client has read access", func() {
+			it("returns true", func() {
+				origImage, err := remote.NewImage(repoName, authn.DefaultKeychain)
+				h.AssertNil(t, err)
+				h.AssertNil(t, origImage.Save())
+
+				image, err := remote.NewImage(repoName, authn.DefaultKeychain)
+				h.AssertNil(t, err)
+
+				h.AssertEq(t, image.CheckReadAccess(), true)
+			})
+		})
+
+		when("image does not exist in the registry and client has read access", func() {
+			it("returns true", func() {
+				image, err := remote.NewImage(repoName, authn.DefaultKeychain)
+				h.AssertNil(t, err)
+				h.AssertEq(t, image.CheckReadAccess(), true)
+			})
+		})
+
+		when("image does not exist in the registry and client hasn't read access", func() {
+			it.Before(func() {
+				os.Setenv("DOCKER_CONFIG", readonlyDockerRegistry.DockerDirectory)
```

Comment:
Maybe this is just the variable name, but I find this confusing - shouldn't the client still have read access to the "read only" registry?

Reply:
When I switched to the read-only registry my auth.DefaultKeyChain is still pointing to dockerRegistry user/pass configuration. in this case, I am trying to access an image that doesn't exist in the registry with invalid credentials, that's why I do not have read access. 

I agreed, maybe the variable name is not appropriated, but it was already there, I didn't want to create another registry instance. Maybe I can point to the same registry with another variable name? could it make more clear?

Reply:
How about just `os.Unsetenv`?

Reply:
Oh! that one seems to be even better!!! Thanks!!!

Reply:
> How about just `os.Unsetenv`?

I replaced the `it.Before` with the `os.Unsetenv` in the following [commit](https://github.com/buildpacks/imgutil/pull/123/commits/03a54599773fab9cb4690816c4dbe72252711908)